backend/api/simulation.py

```
@@ -1766,6 +1793,153 @@ def is_main_character_submit(persona_name, student_role):
+                    scene_progress = db.query(SceneProgress).filter(
+                        and_(
+                            SceneProgress.user_progress_id == user_progress.id,
+                            SceneProgress.scene_id == current_scene_id
+                        )
+                    ).first()
+                    
+                    if scene_progress:
+                        scene_progress.status = "completed"
+                        scene_progress.goal_achieved = False  # Timeout means goal not achieved
+                        scene_progress.forced_progression = True
+                        scene_progress.completed_at = datetime.utcnow()
```

Ruff (0.14.1)

`datetime.datetime.utcnow()` used

(DTZ003)